tests/basilisp/test_core_macros.lpy

```diff
@@ -1,9 +1,32 @@
 (ns tests.basilisp.test-core-macros
-  (:import contextlib inspect os socket tempfile)
+  (:import asyncio contextlib inspect os socket tempfile)
   (:require
    [basilisp.string :as str]
    [basilisp.test :refer [deftest is are testing]]))
 
+(deftest fn-decorators-test
+  (testing "decorators"
+    (testing "in form meta"
+      (let [add-5% #(fn [] (+ (%) 5))
+            f1  ^{:decorators [add-5%]} (fn [] 7)]
+        (is (= 12 (f1)))))
+
+    (testing "in fn name meta"
+      (let [add-5% #(fn [] (+ (%) 5))
+            f2  (fn ^{:decorators [add-5%]} a-fn [] 13)]
+        (is (= 18 (f2)))))
+
+    (testing "with single arg"
+      (let [add-x% (fn [x] #(fn [] (+ (%) x)))
+            f3  (fn ^{:decorators [(add-x% 10)]} _f3 [] 7)]
+        (is (= 17 (f3)))))
```

Comment:
```suggestion
    (testing "in fn name meta"
      (let [add-5% #(fn [] (+ (%) 5))
            f2  (fn ^{:decorators [add-5%]} f2 [] 13)]
        (is (= 18 (f2)))))

    (testing "with single arg"
      (let [add-x% (fn [x] #(fn [] (+ (%) x)))
            f3  (fn ^{:decorators [(add-x% 10)]} f3 [] 7)]
        (is (= 17 (f3)))))
```

I don't think there should be any isues with using the same name for the function as you use in the `let` binding. The function name is useful for diagnostics and (separately) if you want to recurse.

Reply:
You are right indeed!